macrostep.el

```diff
@@ -602,14 +602,17 @@ behaviors."
                 (macrostep-collapse-overlays-in (point) end)
                 (delete-region (point) end)
                 ;; Create a new overlay
-                (let ((overlay
-                       (make-overlay start
-                                     (if (looking-at "\n")
-                                         (1+ (point))
-                                       (point)))))
+                (let* ((overlay
+                        (make-overlay start
+                                      (if (looking-at "\n")
+                                          (1+ (point))
+                                        (point))))
+                       (back-overlay (unless macrostep-expansion-buffer
+                                       (copy-overlay overlay))))
                   (unless macrostep-expansion-buffer
```

Comment:
So if I understand the patch, for each "normal" overlay we additionally create a duplicate overlay with the same extents and priority 0, which is used only for applying the highlight face. I think I'd prefer to use `highlight-overlay` rather than `back-overlay` for both the variable and the property, to make its purpose clearer. Also, the property symbol should be prefixed with `macrostep-` to keep it scoped to this package (so `macrostep-highlight-overlay`). Would you mind making those changes? Otherwise I can do it.